components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -1712,6 +1745,34 @@ public static boolean jsFunction_updateAPI(Context cx, Scriptable thisObj,
                 // retain the previously uploaded image
                 api.setThumbnailUrl(oldApi.getThumbnailUrl());
             }
+
+            if(thumbUrl != null){
+                try {
+                    URL url = new URL(thumbUrl);
```

Comment:
I noticed that we have this icon handling part in two places. Can we make it a one method and call in both places?